src/setup/Agda/Setup.hs

```
@@ -87,26 +83,26 @@ getAgdaAppDir = do
         False -> getXdgDirectory XdgConfig "agda"
 
 -- | This overrides the 'getDataDir' from ''Paths_Agda''.
---   The base data directory defaults to $XDG_DATA_HOME/agda and can be overwritten
---   by the @Agda_datadir@ environment variable.
---   The data directory is then obtained by appending the version
---   (and optional commit information).
+--   If the @data-in-home@ build flag is enabled, the data directory
+--   is obtained by appending the version (and optional commit information)
+--   to the base data directory, which is @$XDG_DATA_HOME/agda@.
+--   Otherwise, the default data directory is used.
```

I suppose setting `Agda_datadir` still works (when the flag is disabled), does it?
Should maybe setting `Agda_datadir` take precedence over the build flag?
This would be more transparent, I think.

---

> I suppose setting `Agda_datadir` still works (when the flag is disabled), does it?

Yes.

> Should maybe setting `Agda_datadir` take precedence over the build flag?
> This would be more transparent, I think.

I don't think it should. I'd rather keep the two "modes" (standard packaging practices vs. self-contained binaries) separated at build time. The flag is supposed to be an approximation of that, so maybe it should already be named `self-contained` or `runtime-data` or something.

---

In this case, we should alert rather visibly of this *breaking change*: Folks that used to set `Agda_datadir` and compiled without attention to cabal flags will maybe be suprised that `Agda_datadir` is now silently ignored.

---

Well, that was already the case with https://github.com/agda/agda/pull/7719. Should the relevant changelog entry be marked "breaking" or something?

---

> I don't think it should. I'd rather keep the two "modes" (standard packaging practices vs. self-contained binaries) separated at build time.

Do you have strong reasons for this?
I just think the situation would be more robust if `data-in-home` just changed the default location of where `agda --setup` puts the data files, and you can always overwrite it with setting `Agda_datadir`.

---

However, I suggest we implement the logic ourselves, instead of using `Paths_Agda`, so that we can sensibly revert to the XDG directories if the environment variable isn't set, rather than reverting to an unlikely to exist Cabal build directory.